File: ssh/Weapon.rb
#encoding:utf-8

require_relative 'CombatElement'

module Irrgarten

	class Weapon < CombatElement
		
		public
		
			def initialize (p, u)	
				super(p,u)
			end
			
			def attack
				return produce_effect
			end
			
			def to_s
				return "W"+super
			end
	end
end

File: irrgarten/src/irrgarten/Weapon.java
package irrgarten;
/**
<<<<<<< HEAD
 *
=======
>>>>>>> 1fe4584be61467713d9bee7c1973916b27fcf79a
 * @author Juan Molina Molina
 * @author Paula Jiménez Ruiz
 * @brief La clase Weapon representa las armas que cada jugador puede utilizar durante un combate
 */
class Weapon extends CombatElement{
    
    public Weapon (float p, int u) {
    
        super(p,u);
    }
    
    public float attack () {
    
        return super.produceEffect();
    }
    
    @Override
    public String toString() {
    
        return "W"+super.toString();
    }
    
}
